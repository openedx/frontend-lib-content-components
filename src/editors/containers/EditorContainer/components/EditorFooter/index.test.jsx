import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { useSelector } from 'react-redux';
import { formatMessage } from '../../../../../testUtils';
import { EditorFooter } from '.';

jest.mock('../../hooks', () => ({
  nullMethod: jest.fn().mockName('hooks.nullMethod'),
}));

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('EditorFooter', () => {
  const props = {
    intl: { formatMessage },
    disableSave: false,
    onCancel: jest.fn().mockName('args.onCancel'),
    onSave: jest.fn().mockName('args.onSave'),
    saveFailed: false,
  };
  describe('render', () => {
    test('snapshot: default args (disableSave: false, saveFailed: false)', () => {
      expect(shallow(<EditorFooter {...props} />).snapshot).toMatchSnapshot();
    });
    test('snapshot: save disabled. Show button spinner', () => {
      expect(shallow(<EditorFooter {...props} disableSave />).snapshot).toMatchSnapshot();
    });
    test('snapshot: save failed.  Show error message', () => {
      expect(shallow(<EditorFooter {...props} saveFailed />).snapshot).toMatchSnapshot();
    });

    test('snapshot: show feedback link', () => {
      useSelector.mockReturnValueOnce('problem');
      expect(shallow(<EditorFooter {...props} />).snapshot).toMatchSnapshot();
    });
    test('snapshot: dont show feedback link', () => {
      useSelector.mockReturnValueOnce('not a Problem');
      expect(shallow(<EditorFooter {...props} />).snapshot).toMatchSnapshot();
    });
  });
});
