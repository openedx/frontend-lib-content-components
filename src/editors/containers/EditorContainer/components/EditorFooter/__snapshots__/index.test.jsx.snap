// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorFooter render snapshot: default args (disableSave: false, saveFailed: false) 1`] = `
<div
  className="editor-footer fixed-bottom"
>
  <ModalDialog.Footer
    className="shadow-sm"
  >
    <ActionRow>
      <Button
        aria-label="Discard changes and return to learning context"
        onClick={[MockFunction args.onCancel]}
        variant="tertiary"
      >
        <FormattedMessage
          defaultMessage="Cancel"
          description="Label for cancel button"
          id="authoring.editorfooter.cancelButton.label"
        />
      </Button>
      <Button
        aria-label="Save changes and return to learning context"
        disabled={false}
        onClick={[MockFunction args.onSave]}
      >
        <FormattedMessage
          defaultMessage="Save"
          description="Label for Save button"
          id="authoring.editorfooter.savebutton.label"
        />
      </Button>
    </ActionRow>
  </ModalDialog.Footer>
</div>
`;

exports[`EditorFooter render snapshot: save disabled. Show button spinner 1`] = `
<div
  className="editor-footer fixed-bottom"
>
  <ModalDialog.Footer
    className="shadow-sm"
  >
    <ActionRow>
      <Button
        aria-label="Discard changes and return to learning context"
        onClick={[MockFunction args.onCancel]}
        variant="tertiary"
      >
        <FormattedMessage
          defaultMessage="Cancel"
          description="Label for cancel button"
          id="authoring.editorfooter.cancelButton.label"
        />
      </Button>
      <Button
        aria-label="Save changes and return to learning context"
        disabled={true}
        onClick={[MockFunction args.onSave]}
      >
        <Spinner
          animation="border"
          className="mr-3"
        />
      </Button>
    </ActionRow>
  </ModalDialog.Footer>
</div>
`;

exports[`EditorFooter render snapshot: save failed.  Show error message 1`] = `
<div
  className="editor-footer fixed-bottom"
>
  <Toast
    show={true}
  >
    <FormattedMessage
      defaultMessage="Error: Content save failed. Please check recent changes and try again later."
      description="Error message displayed when content fails to save."
      id="authoring.editorfooter.save.error"
    />
  </Toast>
  <ModalDialog.Footer
    className="shadow-sm"
  >
    <ActionRow>
      <Button
        aria-label="Discard changes and return to learning context"
        onClick={[MockFunction args.onCancel]}
        variant="tertiary"
      >
        <FormattedMessage
          defaultMessage="Cancel"
          description="Label for cancel button"
          id="authoring.editorfooter.cancelButton.label"
        />
      </Button>
      <Button
        aria-label="Save changes and return to learning context"
        disabled={false}
        onClick={[MockFunction args.onSave]}
      >
        <FormattedMessage
          defaultMessage="Save"
          description="Label for Save button"
          id="authoring.editorfooter.savebutton.label"
        />
      </Button>
    </ActionRow>
  </ModalDialog.Footer>
</div>
`;
